src/MSBuild.Abstractions/MSBuildConversionWorkspace.cs

```diff
@@ -68,9 +68,8 @@ public MSBuildConversionWorkspace(ImmutableArray<string> paths, bool noBackup)
                 if (MSBuildHelpers.ConditionToDimensionValues(propertyGroup.Condition, out var dimensionValues))
                 {
                     var name = MSBuildHelpers.GetConfigurationName(dimensionValues);
+                    if (builder.ContainsKey(name)) continue;
```

Comment:
Style-wise we should use `!builder.Containskey(name)` and introduce another level of nesting. But otherwise this looks good!